### DISCUSSION TITLE
Ship to ship combat concept idea

### DISCUSSION THREAD

**Discussion by Dimasw99** [shiptest-ss13/Shiptest#491] [Ideasguy]
The main goal is to allow player who wish to pvp to also seek it out while those who wish a more peaceful playstyle to stay out of it with some suspense. 

The core pillars are going to be defense, offense, stealth and detection. With combat ships possessing good defense/offense cappabilites and civillian ship being great at running away and staying sneaky.

### Defense:

Energy shield: An energy barrier that surrounds the whole ship with its strength depending on its components and power supplied. The execution could be a out of map solution where we just do the math or having shield blocks surround the whole ship and absorb projectiles directly.

Ablative armor: This ranges from a few reinforced walls outside to _expendable crew quarters_. I can imagine that repairing this could be a nightmare so having a machine that takes a snapshot of your ship and rebuilds would be huge quality of lfie.

Point defense: Direct counter against torpedos. Invest more power for better targeting.

Thrusters: The ability to dodge attacks and escape combat. Gas thrusters taking over a much more important role now as combat thrusters. You may choose to use a lot of moles at low temperature for super high impulses but may run out quicker than the other ship using more efficent but less impulse thrusters. Atmosians rejoice!

Ejectable ballast: Throw your precious resources of your ore sile into space to lower your weight and increase your speed. The oppenent gets to see what was dropped and can decide to either continue fighting or catch the freight before it disappears.

### Offense:

Lasers: Energy to ammo conversion with reliable hit chance but lesser damage.

Projectile: Requires materials, fabrication and reloading with better damage.

Rockets: Expensive but reliable. It could be made to penetrate shields but I think the damge should be lower then. Field repairs can be a nightmare with breaches.

Boarding: Counter to powerful shields, low thrust, high PD. Could be a good last resort method when your own ship is getting its ass beaten.

### Stealth:
Crew size: Intended to keep one player ships protected until more join so that they wont lose their combat ship emidetly. 

Size: This will basically be the innate trait of each ship.

Power consumption: The more machines, lights, RnD stuff, and other crap you run the brighter of a star you will be on the starmap. This could create interresting scenarios where a crew rushes to disable all their APC's to avoid detection of a pirate ship while it passes.

Cloaking: Hide your heatsignature by pumping all heat into dedicated gas storage heating them up. Don't stay hidden for too long or your pipes end up bursting and you end up cooking. Coolers and heat exchangers should be disabled.

### Detection:
Used for both the starmap and combat.

Passive detection: This is our abillity to detect other ships while staying hidden.
 - Cyrostablisied sensors. The cooler the gas the better your detection. The hotter the worse. Could tie into cloaking reducing your accuracy.

Active detection: Sometimes you simply want to see **EVERYTHING** so you become the torch to see in the dark. But everyone else will see you too and even outside your active detection.
 - Sensor burst capacitors. Charge and scan.
 - Radar. Used to get good accuracy during combat while becoming slightly less stealthy.

### Misc:

- Weapons require a warm up/cooldown time when they are being turned on/off to prevent insta stealth with combat ships.
- Being landed on a planet boosts your stealth.
- Initiating combat traps both combetants in a grav field. The better thrusters you have the quicker you can escape it and thus the combat. Gravity generators will fail on both ships for a period amount of time giving us a combat cooldown.
- Energy management should become a main role where a crew will decide which system to supply with more power and for how long your SMES will last. 
- A ship radio announcer should say when a certain ship comes into sensor range with their threat class identification.
- A counter for the pirates would be peacekeeper ships. These would deploy active sensor identifiers giving them the lowest stealth and someone to run to when the baddies appear. They upkeep **SPACE LAW**.
- SoS signals light you up by a significant amount. Ignores planet stealthing
- Third party ship may enter combat too either to aid or contribute to the chaos.

Hopefully some of these ideas spark other better or fitting ideas as trying to implement this whole would require a lot of resources and _shiptesting_. Had lots of fun on your server and thanks for reading, sincearly DimasW (pls don't confuse me with the other dimas).

**Comment by ZephyrTFA** [comment]
> The main goal is to allow player who wish to pvp to also seek it out while those who wish a more peaceful playstyle to stay out of it with some suspense.
> 
> The core pillars are going to be defense, offense, stealth and detection. With combat ships possessing good defense/offense cappabilites and civillian ship being great at running away and staying sneaky.
> 
> ### Defense:
> Energy shield: An energy barrier that surrounds the whole ship with its strength depending on its components and power supplied. The execution could be a out of map solution where we just do the math or having shield blocks surround the whole ship and absorb projectiles directly.
> 
> Ablative armor: This ranges from a few reinforced walls outside to _expendable crew quarters_. I can imagine that repairing this could be a nightmare so having a machine that takes a snapshot of your ship and rebuilds would be huge quality of lfie.
> 
> Point defense: Direct counter against torpedos. Invest more power for better targeting.
> 
> Thrusters: The ability to dodge attacks and escape combat. Gas thrusters taking over a much more important role now as combat thrusters. You may choose to use a lot of moles at low temperature for super high impulses but may run out quicker than the other ship using more efficent but less impulse thrusters. Atmosians rejoice!
> 
> ### Offense:
> Lasers: Energy to ammo conversion with reliable hit chance but lesser damage.
> 
> Projectile: Requires materials, fabrication and reloading with better damage.
> 
> Rockets: Expensive but reliable. It could be made to penetrate shields but I think the damge should be lower then. Field repairs can be a nightmare with breaches.
> 
> ### Stealth:
> Crew size: Intended to keep one player ships protected until more join so that they wont lose their combat ship emidetly.
> 
> Size: This will basically be the innate trait of each ship.
> 
> Power consumption: The more machines, lights, RnD stuff, and other crap you run the brighter of a star you will be on the starmap. This could create interresting scenarios where a crew rushes to disable all their APC's to avoid detection of a pirate ship while it passes.
> 
> ### Detection:
> Passive detection: This is our abillity to detect other ships while staying hidden. Active detection: Sometimes you simply want to see **EVERYTHING** so you become the torch to see in the dark. But everyone else will see you too and even outside your active detection.
> 
> ### Misc:
> * Weapons require a warm up/cooldown time when they are being turned on/off to prevent insta stealth with combat ships.
> * Being landed on a planet boosts your stealth.
> * Initiating combat traps both combetants in a grav field. The better thrusters you have the quicker you can escape it and thus the combat. Gravity generators will fail on both ships for a period amount of time giving us a combat cooldown.
> * Energy management should become a main role where a crew will decide which system to supply with more power and for how long your SMES will last.
> * A ship radio announcer should say when a certain ship comes into sensor range with their threat class identification.
> * A counter for the pirates would be peacekeeper ships. These would deploy active sensor identifiers giving them the lowest stealth and someone to run to when the baddies appear. They upkeep **SPACE LAW**.
> * SoS signals light you up by a significant amount. Ignores planet stealthing
> * Third party ship may enter combat too either to aid or contribute to the chaos.
> 
> Hopefully some of these ideas spark other better or fitting ideas as trying to implement this whole would require a lot of resources and _shiptesting_. Had lots of fun on your server and thanks for reading, sincearly DimasW (pls don't confuse me with the other dimas).

See #434

**Comment by Dimasw99** [comment]
Another Idea addition to the set. Players end up mining sometimes hundrest if not thousands of sheet which are all stored in the ore silo.
During combat you could use **ejectable ballast** to help you survive another day while the pirates still get to have a little booty. Pirates will gain while the loser gets away with his life.

**Reply by Dimasw99** [reply]
EDIT: added Cloaking, Ejectable ballast, boarding and detection components to the list.

**Comment by ilikesaying** [comment]
Another idea - a separate system that contains ai pirate/hostile ships

**Reply by Dimasw99** [reply]
Totally agreed. Having PvE content with hostile ship is both going to benefit lowpop and help get more feedback.

**Comment by VeteranKyl** [comment]
By the sounds of things, this sort of system would be best applied to a combat system similar to the game 'FTL: Faster Than Light'. I think having combat set up in a similar fashion to that game would be the best way of achieving this.

One question I have though is how this system would work with ships that are currently docked. Would a docked ship be forced into a combat map if someone attacks it? What if multiple ships are docked in the same map? Would other ships be able to partake in a battle that's in progress?
